src/components/action-bar/action-bar.tsx

```diff
@@ -0,0 +1,283 @@
+     * :::note
+     * You should still properly position the component
+     * according to the structure of your user interface.
+     * For example, use an `absolute` or `fixed` position.
+     * :::
+     */
+    @Prop({ reflect: true })
+    public layout?: 'fullWidth' | 'floating';
+
+    /**
+     * Fired when a action bar item has been clicked.
+     */
+    @Event()
+    public itemSelected: EventEmitter<ActionBarItem>;
+
+    @Element()
+    public element: HTMLElement;
+
+    /**
+     * Defines the location that the content of the overflow menu
+     * appears, in relation to its trigger.
+     */
+    @Prop({ reflect: true })
+    public openDirection: OpenDirection;
+
+    @State()
+    private overflowCutoff: number = this.actionBarItems?.length || 0;
+
+    @State()
+    private calculatingOverflow: boolean = false;
+
+    private resizeObserver: ResizeObserver;
+    private intersectionObserver: IntersectionObserver;
+    private overflowRecalculationScheduled: boolean;
+    // eslint-disable-next-line no-magic-numbers
+    private RESIZE_RATE_MS = 50;
+    // eslint-disable-next-line no-magic-numbers
+    private RESIZE_TIMEOUT = 250;
+    private lastOverflowCutoff: number = this.actionBarItems?.length || 0;
+    private lastResize = Date.now();
+
+    public componentWillLoad() {
+        this.resizeObserver = new ResizeObserver(this.handleResize);
+    }
+
+    public disconnectedCallback() {
+        this.resizeObserver.disconnect();
+        this.intersectionObserver.disconnect();
+    }
+
+    public render() {
+        let visibleItems: Array<ActionBarItem | ListSeparator> = [];
+        let overflowItems: Array<MenuItem | ListSeparator> = [];
+        if (this.actionBarItems.length) {
+            visibleItems = this.actionBarItems.slice(0, this.overflowCutoff);
+            overflowItems = this.actionBarItems.slice(this.overflowCutoff);
+        }
+
+        return (
+            <Host
+                aria-label={this.accessibleLabel}
+                class={{
+                    'is-full-width': this.layout === 'fullWidth',
+                    'is-floating': this.layout === 'floating',
+                }}
+            >
+                {visibleItems.map(this.renderActionBarItem)}
+                {this.renderOverflowMenu(overflowItems)}
+            </Host>
+        );
+    }
+
+    public componentDidRender() {
+        const element =
+            this.element.parentElement ?? (this.element.parentNode as any).host;
+
+        if (element === null) {
+            throw new Error(
+                'Failed to create resizeObserver, no parent found for limel-action-bar. Consider wrapping the action-bar in an element'
+            );
+        }
+
+        this.resizeObserver.observe(element);
+        if (this.calculatingOverflow) {
+            this.createIntersectionObserver();
+        }
+    }
+
+    @Watch('actionBarItems')
+    protected handleActionBarItemsChange() {
+        setTimeout(this.handleResize, 0);
```

Comment:
I added this. Without it, the bar wouldn't handle being created without items and then having them added very well. I found it while trying to write tests, but it is a legit use case when it's optional to provide items when the component is created (it's also good to keep it optional, as it not straightforward to supply non-primitive values at the time of creation in tests (if it's even possible… 🤔)